lib/fluent/plugin/out_kafka.rb

```diff
@@ -162,7 +162,7 @@ def emit(tag, es, chain)
         value = @formatter_proc.call(tag, time, record)
 
         log.on_trace { log.trace("message send to #{topic} with key: #{partition_key} and value: #{value}.") }
-        producer.produce(value, topic: topic, partition_key: partition_key)
+        producer.produce(value, topic: topic, key: partition_key)
```

Comment:
Ruby-kafka uses message_key for partitioning if partition_key is not set explicitly. 
Can you check [Ruby-kafka Usage](https://github.com/zendesk/ruby-kafka#producing-messages-to-kafka)